src/backend/database_models/agent_tool_metadata.py

```diff
@@ -0,0 +1,29 @@
+from typing import Optional
+
+from sqlalchemy import ForeignKey, Text, UniqueConstraint
+from sqlalchemy.dialects.postgresql import ARRAY
+from sqlalchemy.orm import Mapped, mapped_column
+
+from backend.database_models.base import Base
+
+
+class AgentToolMetadata(Base):
+    __tablename__ = "agent_tool_metadata"
+
+    user_id: Mapped[str] = mapped_column(Text, nullable=False)
+    agent_id: Mapped[str] = mapped_column(
+        ForeignKey("agents.id", ondelete="CASCADE"), nullable=False
+    )
+    tool_name: Mapped[str] = mapped_column(Text, nullable=False)
+    type: Mapped[str] = mapped_column(Text, nullable=False)
+    artifacts: Mapped[list[str]] = mapped_column(ARRAY(Text), nullable=True)
```

Comment:
So this implies that we have one type for a list of artifacts? I think it's a good design.
Can also have pairs of (type, artifact_id) and have a uc on "user_id", "agent_id", "tool_name", "type", "artifact_id". Curious what @BeatrixCohere thinks as well.

Reply:
yeah it would. can't predict the future but i would imagine most use cases would be strings. 

as to storing as pairs instead:
I'm not too sure how I feel about that. I noticed when writing all this that keep the fields as basic types makes it a lot easier to read and parse. I also think the API payloads should be close if not the same as the DB schema for simplicity of development, and if we introduce a composite type here (pairs) then we'd probably want to parse pairs in the payload, more checks, etc, etc.

open to change tho, so if you feel strongly otherwise we can discuss!